include/openssl/kdf.h

```diff
@@ -90,6 +93,19 @@ extern "C" {
             EVP_PKEY_CTX_ctrl_uint64(pctx, -1, EVP_PKEY_OP_DERIVE, \
                             EVP_PKEY_CTRL_SCRYPT_MAXMEM_BYTES, maxmem_bytes)
 
+# define EVP_PKEY_CTX_set_pbkdf2_md(pctx, md) \
+            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
+                              EVP_PKEY_CTRL_PBKDF2_MD, 0, (void *)(md))
+
+# define EVP_PKEY_CTX_set1_pbkdf2_salt(pctx, salt, saltlen) \
```

Comment:
Hmmm, top of my head, I would think that a `EVP_PKEY_CTRL_KDF_` could be a common prefix if that's needed.  However, the ctrl functions for the different implementations (scrypt, pbkdf2, ...) don't care exactly what command numbers are thrown at them, as long as each number is unique for that implementation.  For all hkdf, scrypt and pbkdf2 would care, the command macros used for each of them could hve overlapping numberings, they wouldn't even notice.

In *my* mind, it would be perfectly ok to reuse the already defined `EVP_PKEY_CTRL_MD`, but for most humans, it might be better to have something that's at least associated with KDFs, so perhaps `EVP_PKEY_CTRL_KDF_MD`, `EVP_PKEY_CTRL_KDF_PASS`, `EVP_PKEY_CTRL_KDF_SALT`, etc etc etc.  You follow?